### ISSUE TITLE
Rapid tests are not saved after 3rd vaccine

### ISSUE THREAD

**Issue by CallMeFlanby** [corona-warn-app/cwa-app-android#4585]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

## Technical details

- Device name: Samsung Galaxy Note 10 (SM-N970F/DS)
- Android version: Android 11
- App version: 2.14

## Describe the bug

Since I added my last vaccine (the 3rd), the rapid tests aren't visible anymore in the menu "Zertifikate".

## Steps to reproduce the issue

1. Go to the menu "Status" 
2. Click on the last rapid test

![Screenshot_20211218-143943_Corona-Warn.jpg](https://user-images.githubusercontent.com/7440579/146643104-396c49c1-5ffc-4786-9acc-0b3fd72d4625.jpg)
3. Check that the current rapid test is valid

![Screenshot_20211218-143949_Corona-Warn.jpg](https://user-images.githubusercontent.com/7440579/146643117-800f2dd3-fa66-45e8-ab64-b3af3c14a67c.jpg)
4. Go now back and to the menu "Zertifikate"
5. The current rapid test is now shown as a valid certificate

![Screenshot_20211218-144001_Corona-Warn.jpg](https://user-images.githubusercontent.com/7440579/146643138-2847ecd2-c69f-427d-8bca-331d638c9e0f.jpg)

## Expected behaviour

Rapid tests should be shown in the certificate menu also after the 3rd vaccine.

## Additional context

At least two tests aren't shown currently. Before I was vaccinated (on the 14th December) I had a rapid test that I added to the app. Since I added the last vaccine however, this test disappeared. The test from today is also not visible within the menu "Zertifikate". However, if I try to add the tests again CWA tells me they are already registered. In the rapid test details there is no QR Code, which makes (at least for me) useless to add negative rapid tests to CWA as I cannot show them anywhere.

**Comment by Ein-Tim** [comment]
1. Your 3/3 certificate is valid, there is no 14 day waiting period
2. What exactly are you missing? I can see one test certificate from 13.12. is displayed in the app and one from 09.12.

**Comment by CallMeFlanby** [comment]
> 1. Your 3/3 certificate is valid, there is no 14 day waiting period.
> 2. What exactly are you missing? I can see one test certificate from 13.12. is displayed in the app and one from 09.12. Are these the two tests you are "Missing"?

1. Sorry, thought that the 14 day period was for everyone. I live in MV and we have to wait 14 days (https://www.ndr.de/nachrichten/mecklenburg-vorpommern/Keine-Testpflicht-mehr-nach-Booster-Impfung-in-MV,coronavirus6158.html). Thanks, I'll update the issue.
2. I had a rapid test on the 14th and on the 18th December. Both of them are not visible.

**Comment by Ein-Tim** [comment]
> Sorry, thought that the 14 day period was for everyone. I live in MV and we have to wait 14 days (ndr.de/nachrichten/mecklenburg-vorpommern/Keine-Testpflicht-mehr-nach-Booster-Impfung-in-MV,coronavirus6158.html).

Ah okay, I was not aware of this regulation in MV. I guess you can already now use your booster vaccination to get entry, however, you still have to get tested for 14 days if you want to visit a 2G+ event until 14 days since the vaccination have passed. l

> I had a rapid test on the 14th and on the 18th December. Both of them are not visible.

Are you sure that you got a test certificate (QR-Code) for them? Where the tests only shown on the "Status" tab or also on the "Certificates" tab. Did you take a look at the recycle bin to see whether they were deleted?

**Comment by CallMeFlanby** [comment]
> > I had a rapid test on the 14th and on the 18th December. Both of them are not visible.
> 
> Are you sure that you got a test certificate (QR-Code) for them? Where the tests only shown on the "Status" tab or also on the "Certificates" tab. Did you take a look at the recycle bin to see whether they were deleted? 

I  just checked my trash bin and found out that the test from  the 14th December was there, however it was never shown in the "Certificates" tab. 

Here a screenshot of my trash bin

![Screenshot_20211218-172414_Corona-Warn.jpg](https://user-images.githubusercontent.com/7440579/146648510-7463c95e-9f18-4aa2-9dfb-d716c31b6ac0.jpg)

Has you can see, the test from today (18.12) is visible in the "Status" tab but not in the "certificates" tab. 

Status tab:

![Screenshot_20211218-172419_Corona-Warn.jpg](https://user-images.githubusercontent.com/7440579/146648554-6f308320-b15b-4b38-b1e5-e9a40a7cba64.jpg)


Certificate tab:

![Screenshot_20211218-172428_Corona-Warn.jpg](https://user-images.githubusercontent.com/7440579/146648564-b06ae458-4a58-4ec8-9abd-faee8092250b.jpg)


Do you know what I mean now? 😊

**Comment by CallMeFlanby** [comment]
I remembered now that I did something different as I was adding the rapid test. 

![Screenshot_20211218-173721_Corona-Warn.jpg](https://user-images.githubusercontent.com/7440579/146648751-b69fee89-8014-4f2a-b77e-5b4d133cbbdb.jpg)

As I was in this window, I chose "nein danke". Is it maybe related with it?

**Comment by Ein-Tim** [comment]
@CallMeFlanby 

Yes exactly, you can have a test registered in the app (shown on the status tab) but if you declined to receive a test certificate, you won't get one. 

I will try to explain it, there are two different things related to tests in the app.

1. Test results
	They are shown on the "Status" tab and you can always only have one PCR and one rapid antigen test result stored in the app at the same time.
2. Test certificates
	They are shown on the certificates tab. If you register a **test result** in your app, it is possible to request a **test certificate**. However, some testing stations do not offer the creation of test results **and** you can decline the creation of a test certificate while registering the **test result** in your app.

So, in conclusion, there can be a test result registered in the app (shown on the "Status" tab) but it's possible that for this test result not test certificate is shown (on the "Certificates" tab). 
	
I hope you understand what I mean?

**Comment by CallMeFlanby** [comment]
Thank you very much. Closing the issue.
